File: dodam-api/src/main/java/b1nd/dodamapi/auth/handler/AuthController.java
package b1nd.dodamapi.auth.handler;

import b1nd.dodamapi.auth.usecase.AuthUseCase;
import b1nd.dodamapi.common.response.ResponseData;
import b1nd.dodamcore.auth.application.dto.req.LoginReq;
import b1nd.dodamcore.auth.application.dto.req.ReissueTokenReq;
import b1nd.dodamcore.auth.application.dto.res.LoginRes;
import b1nd.dodamcore.auth.application.dto.res.ReissueTokenRes;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.concurrent.CompletableFuture;

@RestController
@RequestMapping("/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthUseCase useCase;

    @PostMapping("/login")
    public CompletableFuture<ResponseData<LoginRes>> login(@RequestBody @Valid LoginReq req) {
        return useCase.login(req);
    }

    @PostMapping("/reissue")
    public CompletableFuture<ResponseData<ReissueTokenRes>> reissue(@RequestBody @Valid ReissueTokenReq req) {
        return useCase.reissue(req);
    }

}

File: services/service-auth/src/main/kotlin/com/b1nd/dodamdodam/auth/presentation/auth/AuthController.kt
package com.b1nd.dodamdodam.auth.presentation.auth

import com.b1nd.dodamdodam.auth.application.auth.AuthUseCase
import com.b1nd.dodamdodam.auth.application.auth.data.request.LoginRequest
import com.b1nd.dodamdodam.auth.application.auth.data.response.LoginResponse
import com.b1nd.dodamdodam.core.common.data.Response
import com.b1nd.dodamdodam.core.security.annotation.authentication.UserAccess
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class AuthController(
    private val useCase: AuthUseCase
) {
    @PostMapping("/login")
    fun login(@RequestBody request: LoginRequest): Response<LoginResponse> =
        useCase.login(request)

    @GetMapping("/health")
    fun health(): String = "OK"

    @UserAccess(enabledOnly = true)
    @GetMapping("/test")
    fun test(): String = "TEST"
}